scripts/apps/authoring-react/manage-editor3-inside-authoring-react.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import {registerInternalExtension} from 'core/helpers/register-internal-extension';
+import {
+    IExtensionActivationResult,
+    IEditorComponentProps,
+    ICustomFieldType,
+} from 'superdesk-api';
+import {gettext} from 'core/utils';
+import {RawDraftContentState, convertToRaw, ContentState} from 'draft-js';
+import createEditorStore, {IEditorStore} from 'core/editor3/store';
+import ng from 'core/services/ng';
+import {Provider} from 'react-redux';
+import {Store} from 'redux';
+import {Editor3} from 'core/editor3/components';
+
+const EditorStore = React.createContext<Store>(null);
+
+interface IValue {
+    store: Store<IEditorStore>;
+    contentState: ContentState;
+}
+
+type IProps = IEditorComponentProps<IValue, unknown>;
+
+class Editor3Component extends React.PureComponent<IProps> {
+    render() {
+        const store = this.props.value.store;
+
+        /**
+         * TODO: get configs from content profile and pass them as props
+         */
+
+        return (
+            <Provider store={store}>
+                <EditorStore.Provider value={store}>
+                    <Editor3
+                        scrollContainer="window"
+                        singleLine={false}
+                        cleanPastedHtml={false}
+                        autocompleteSuggestions={undefined}
+                    />
+                </EditorStore.Provider>
+            </Provider>
+        );
+    }
+}
+
+const editor3AuthoringReact = 'editor3--authoring-react';
+
+export function registerEditor3AsCustomField() {
+    /**
+     * TODO: add preview component and all missing data for store initialization.
+     */
+    const customFields: Array<ICustomFieldType<IValue, unknown, RawDraftContentState>> = [
+        {
+            id: 'editor3',
+            label: gettext('Editor3 (authoring-react)'),
+            editorComponent: Editor3Component,
+            // previewComponent: Editor3Component
+            previewComponent: () => null,
+
+            fromStorageValue: (rawContentState, config, onChange) => {
```

Comment:
I'm not sure if this interface will be enough, the editor might need also associations

Reply:
I've changed the API a bit since then. Here's the new one

```
storeValue?(fieldId: string, article: IArticle, value: IValue): IArticle;
retrieveStoredValue?(fieldId: string, article: IArticle): IValue;
```

If associations are needed, `retrieveStoredValue` could be implemented as follows:
```
retrieveStoredValue: (fieldId, article) => {
    const rawContentState = article.fields_meta?.[fieldId]?.['draftjsState'][0];

    const store = createEditorStore(
        {
            editorState: rawContentState ?? convertToRaw(ContentState.createFromText('')),
            // ...
        },
        ng.get('spellcheck'),
        true,
    );

    return {
        store,
        contentState: store.getState().editorState.getCurrentContent(),
        associations: getAssociations(article),
    };
},
```

It might only be a challenge if 2 or more fields wanted to use associations, then we might consider passing it directly, but until now current interface would work in my opinion.